presto-raptorx/src/main/java/com/facebook/presto/raptorx/RaptorMetadata.java

```diff
@@ -0,0 +1,743 @@
+                            .map(RaptorMetadata::toColumnMetadata)
+                            .collect(toImmutableList())));
+        }
+        return map.build();
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, boolean ignoreExisting)
+    {
+        Optional<ConnectorNewTableLayout> layout = getNewTableLayout(session, tableMetadata);
+        finishCreateTable(session, beginCreateTable(session, tableMetadata, layout), ImmutableList.of(), ImmutableList.of());
+    }
+
+    @Override
+    public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        transaction.dropTable(((RaptorTableHandle) tableHandle).getTableId());
+    }
+
+    @Override
+    public void renameTable(ConnectorSession session, ConnectorTableHandle tableHandle, SchemaTableName newTableName)
+    {
+        if (!schemaExists(session, newTableName.getSchemaName())) {
+            throw new SchemaNotFoundException(newTableName.getSchemaName());
+        }
+        transaction.renameTable(((RaptorTableHandle) tableHandle).getTableId(), newTableName);
+    }
+
+    @Override
+    public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnMetadata column)
+    {
+        transaction.addColumn(
+                ((RaptorTableHandle) tableHandle).getTableId(),
+                column.getName(),
+                column.getType(),
+                Optional.ofNullable(column.getComment()));
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle source, String target)
+    {
+        long tableId = ((RaptorTableHandle) tableHandle).getTableId();
+        long columnId = ((RaptorColumnHandle) source).getColumnId();
+        transaction.renameColumn(tableId, columnId, target);
+    }
+
+    @Override
+    public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        long tableId = ((RaptorTableHandle) tableHandle).getTableId();
+        long columnId = ((RaptorColumnHandle) columnHandle).getColumnId();
+
+        TableInfo table = transaction.getTableInfo(tableId);
+
+        if (table.getTemporalColumnId().isPresent() && (table.getTemporalColumnId().getAsLong() == columnId)) {
+            throw new PrestoException(NOT_SUPPORTED, "Cannot drop the temporal column");
+        }
+
+        ColumnInfo column = table.getColumns().stream()
+                .filter(info -> info.getColumnId() == columnId)
+                .collect(onlyElement());
+
+        if (column.getBucketOrdinal().isPresent()) {
+            throw new PrestoException(NOT_SUPPORTED, "Cannot drop bucket columns");
+        }
+        if (column.getSortOrdinal().isPresent()) {
+            throw new PrestoException(NOT_SUPPORTED, "Cannot drop sort columns");
+        }
+
+        transaction.dropColumn(tableId, columnId);
```

Comment:
This isn't needed since we find the column metadata for this column ID a few lines up. If the engine called it with a hidden column, the `onlyElement()` collector would throw.